File: app/src/main/go/utilities.go
package smsmatrix

import (
	"log"
	"os/exec"
)

// SendMatrix sends an SMS event to MatrixService.
// This must be an exported function (capitalized) to be accessible in Java/Kotlin.
func SendMatrix(phone, body, msgType, fileName, contentType string) {
	log.Println("Arrived in SendMatrix")

	cmd := exec.Command("/system/bin/sh", "-c",
		`am start-foreground-service -a com.averydorgan.smsmatrix.SEND_MATRIX -n com.averydorgan.smsmatrix/.MatrixService --es SendSms_phone "`+phone+`" --es SendSms_body "`+body+`" --es SendSms_type "`+msgType+`" --es SendSms_fileName "`+fileName+`" --es SendSms_contentType "`+contentType+`"`,
	)

	log.Println("Running SendMatrix Command")
	if err := cmd.Run(); err != nil {
		log.Println("Failed to start MatrixService:", err)
	}
}

File: app/src/main/java/eu/droogers/smsmatrix/Utilities.kt
package eu.droogers.smsmatrix

import android.content.Context
import android.content.Intent
import eu.droogers.smsmatrix.MatrixService

object Utilities {
    @JvmStatic
    fun sendMatrix(context: Context, body: String?, phone: String?, type: String?) {
        val intent = Intent(context, MatrixService::class.java)
        intent.putExtra("SendSms_phone", phone)
        intent.putExtra("SendSms_body", body)
        intent.putExtra("SendSms_type", type)
        context.startService(intent)
    }

    fun sendMatrix(
        context: Context,
        body: ByteArray?,
        phone: String?,
        type: String?,
        fileName: String?,
        contentType: String?
    ) {
        val intent = Intent(context, MatrixService::class.java)
        intent.putExtra("SendSms_phone", phone)
        intent.putExtra("SendSms_body", body)
        intent.putExtra("SendSms_type", type)
        intent.putExtra("SendSms_fileName", fileName)
        intent.putExtra("SendSms_contentType", contentType)
        context.startService(intent)
    }
}
